mcv/hostname.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+
+def _read_stripped_lines(f):
+    return [l.rstrip() for l in f.readlines()]
+
+
+def _splice_alias(alias, line):
+    """If line is an alias for localhost, append <alias>
+    (if not already present)"""
+    vs = line.split(' ')
+    if vs[0] == '127.0.0.1' and not alias in vs[1:]:
+        vs.append(alias)
+        return ' '.join(vs)
+    else:
+        return line
+
+
+def set_hostname(hostname):
+    """Set the system hostname to <hostname> and add a
+    corresponding entry to /etc/hosts"""
+    os.system('hostname ' + hostname)
+    with open('/etc/hostname', 'w') as f:
+        f.write(hostname)
+    with open('/etc/hosts', 'r+w') as f:
```

Comment:
I think this is not too far off from that - since we're reading and writing to the same file, you don't really have a choice about the seek ?